tests/psp1155.tests.ts

```diff
@@ -0,0 +1,257 @@
+        const { query, contract, defaultSigner: sender } = await setup()
+
+        let token1 = 1
+        let token2 = 2
+        let amount1 = 1
+        let amount2 = 10
+
+        const BOB = await getSigner('Bob')
+
+        await contract.tx.mint_to(sender.address, token1, amount1)
+        await contract.tx.mint_to(sender.address, token2, amount2)
+        await contract.tx.mint_to(BOB.address, token1, amount1)
+        await contract.tx.mint_to(BOB.address, token2, amount2)
+
+        await contract.tx.burn_batch([token1, token2], [amount1, amount2], []);
+        await expect(() => contract.tx.balance_of_batch([sender.address, sender.address], [token1, token2]))
+            .to.have.output([0, 0])
+
+        await fromSigner(contract, BOB.address).tx.set_approval_for_all(sender.address, true)
+        await contract.tx.burn_batch_from(BOB.address, [token1, token2], [amount1, amount2], []);
+        await expect(() => contract.tx.balance_of_batch([BOB.address, BOB.address], [token1, token2]))
+            .to.have.output([0, 0])
+    })
+
+    it('Burn from without allowance should fail', async () => {
+        const { query, contract, defaultSigner: sender } = await setup()
+
+        let token1 = 1
+        let token2 = 2
+        let amount1 = 1
+        let amount2 = 10
+
+        const BOB = await getSigner('Bob')
+
+        await contract.tx.mint_to(BOB.address, token1, amount1)
+        await contract.tx.mint_to(BOB.address, token2, amount2)
+
+        await expect(() => contract.tx.burn_batch_from(BOB.address, [token1, token2], [amount1, amount2], []))
+            .to.eventually.be.rejected
+        await expect(() => contract.tx.burn_from(BOB.address, token1, amount1, []))
+            .to.eventually.be.rejected
+    })
+
+    it('Burn from inssuficient balance should fail', async () => {
+        const { query, contract, defaultSigner: sender } = await setup()
+
+        let token1 = 1
+        let token2 = 2
+        let amount1 = 1
+        let amount2 = 10
+
+        const BOB = await getSigner('Bob')
+
+        await contract.tx.mint_to(sender.address, token1, amount1)
+        await contract.tx.mint_to(sender.address, token2, amount2)
+        await contract.tx.mint_to(BOB.address, token1, amount1)
+        await contract.tx.mint_to(BOB.address, token2, amount2)
+
+        await expect(() => contract.tx.burn_batch([token1 + 1, token2], [amount1, amount2], []))
+            .to.eventually.be.rejected
+        await expect(() => contract.tx.burn(token1 + 1, amount1, []))
+            .to.eventually.be.rejected
+
+        await fromSigner(contract, BOB.address).tx.set_approval_for_all(sender.address, true)
+        await expect(() => contract.tx.burn_batch_from(BOB.address, [token1, token2], [amount1 + 1, amount2], []))
+            .to.eventually.be.rejected
+        await expect(() => contract.tx.burn_from(BOB.address, token1, amount1 + 1, []))
+            .to.eventually.be.rejected
+
+    })
+})
```

Comment:
Can you rearrange your code to match AAA pattern? Have a look at ownable.tests.ts